### PR TITLE
VB-77 Fileuploader, show error message

### PR DESCRIPTION
**Pull request by assi98**
Small change. Alert the user when they try to upload a to big receipe file.

### CODE REVIEW

**Review by Lundli — CHANGES_REQUESTED** [review]
This change seems to break the styling/css of the utlegg-page.
The styling disappears when the page is reloaded.

image


See commit f948c82e4f911f8a55f1f2ca50a4ea0e3982a0ac which fixes the form layout.

**Comment by Lundli** [comment]
When notifying the user about the error (image too large), I would suggest you use `addFlash` to be consistent with other pages.
`$this->addFlash('warning', "text here");`

It would then look like this:
image

Example can be found here:
https://github.com/vektorprogrammet/vektor-backend/blob/18a24c51ebcf444e1403d3e533ecc885ea72f374/src/Controller/SemesterController.php#L56

**Comment by assi98** [comment]
Are now using flash note and it seem to have fixed css issues: 
image

File: src/Controller/ReceiptController.php [hunk 1]
```diff
@@ -118,16 +122,19 @@ public function create(Request $request)
...
         }
         // Else: User is viewing receipt page (no receipt exist: path=null)
-        else {
-            echo "<script>alert('The file is too large. Allowed maximum size is 2 MiB.)";
-            $receipt->setPicturePath("");
+        else{ $receipt->setPicturePath("");}
+
+        if ($form->isSubmitted() && !$form->isValid()&& $receipt->getPicturePath()=="") {
+
+            $this->addFlash('warning', "The file is too large. Allowed maximum size is 2 MiB.)");
```
Comment (Lundli) [inline src/Controller/ReceiptController.php:129]: Rememeber that this page is in norwegian. Error messages shown to user should be as well.

File: src/Controller/ReceiptController.php [hunk 2]
```diff
@@ -118,15 +122,19 @@ public function create(Request $request)
             return $this->redirectToRoute('receipt_create');
         }
         // Else: User is viewing receipt page (no receipt exist: path=null)
-        else {
-            $receipt->setPicturePath("");
+        else{ $receipt->setPicturePath("");}
+
+        if ($form->isSubmitted() && !$form->isValid()&& $receipt->getPicturePath()=="") {
```
Comment (Lundli) [inline src/Controller/ReceiptController.php:127]:
```suggestion
        if ($form->isSubmitted() && !$form->isValid()&& $receipt->getPicturePath() == "") {
```

File: package-lock.json [hunk 3]
```diff
@@ -12100,238 +12100,273 @@
     }
   },
   "dependencies": {
+    "@ampproject/remapping": {
```
Comment (Lundli) [inline package-lock.json:12103]: Looks great! :100: 

For information:
We have moved from `npm` to `yarn`.
You should get new changes from upstream, and re-create lock files using yarn (`yarn.lock`).

**TODOs:**
- [ ] Merge incoming upstream changes
- [ ] Fix suggested changes
Should be good after that!

**Review by Lundli — APPROVED** [review]
:100:

**Pull request merged**